src/azure-cli/azure/cli/command_modules/serviceconnector/_utils.py

```diff
@@ -129,3 +130,99 @@ def auto_register(func, *args, **kwargs):
             raise CLIInternalError('Registeration failed, please manually run command '
                                    '`az provider register -n Microsoft.ServiceLinker` to register the provider.')
         raise ex
+
+
+def create_key_vault_reference_connection_if_not_exist(cmd, client, source_id, key_vault_id):
+    from ._validators import get_source_resource_name
+    from knack.log import get_logger
+    logger = get_logger(__name__)
+
+    logger.warning('get valid key vualt reference connection')
+    all_connections = run_cli_cmd('az webapp connection list --source-id {} -o json'.format(source_id))
+    key_vault_connections = []
+    for connection in all_connections:  # pylint: disable=not-an-iterable
+        if connection.get('targetId') == key_vault_id:
+            key_vault_connections.append(connection)
+
+    source_name = get_source_resource_name(cmd)
+    auth_info = get_auth_if_no_valid_key_vault_connection(logger, source_name, source_id, key_vault_connections)
+    if not auth_info:
+        return
+
+    # No Valid Key Vault Connection, Create
+    logger.warning('no valid key vault connection found. Creating...')
+    client = set_user_token_header(client, cmd.cli_ctx)
```

Comment:
why setting user token again, now that it's already done in https://github.com/Azure/azure-cli/pull/21471/files#diff-36d274d07efd6eddc6fb27a4135f206bb5caa865c9a4b1a4a260ad1a51a31911R200

Reply:
removed.
As set_user_token_header is not a heavy operation. It could avoid potential error if others call it, actually.